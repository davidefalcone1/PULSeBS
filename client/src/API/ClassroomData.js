class ClassroomData{
    constructor(id, classroomName, maxSeats) {
        this.classId = id;
        this.classroomName = classroomName;
        this.maxSeats = maxSeats;
    }

    static fromJson(json){
        const temp =  Object.assign(new ClassroomData(), json);
        return temp;
    }
}

export default ClassroomData;