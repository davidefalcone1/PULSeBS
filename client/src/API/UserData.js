class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;