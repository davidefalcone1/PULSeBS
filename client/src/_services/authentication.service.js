import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}