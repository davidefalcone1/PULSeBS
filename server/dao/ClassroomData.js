class ClassroomData{
    constructor(id, classroomName, maxSeats) {
        this.classId = id;
        this.classroomName = classroomName;
        this.maxSeats = maxSeats;
    }

}

module.exports = ClassroomData;