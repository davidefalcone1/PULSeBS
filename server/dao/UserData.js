class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;