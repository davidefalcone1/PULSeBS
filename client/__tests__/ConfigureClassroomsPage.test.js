import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import ConfigureClassroomsPage from '../src/components/ConfigureClassroomsPage';
import { BrowserRouter } from 'react-router-dom';
import ClassroomData from '../src/API/ClassroomData';
import UserData from '../src/API/UserData';

describe('ConfigureClassroomsPage', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}}>
                    <ConfigureClassroomsPage classesList={[new ClassroomData(1, 'A1', 70)]} createNewClass={()=>{}} uploadFileClassrooms={()=>{}}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});