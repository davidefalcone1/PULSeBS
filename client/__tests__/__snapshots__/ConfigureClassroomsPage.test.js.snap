// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureClassroomsPage Renders correctly 1`] = `
Array [
  <br />,
  <div
    className="justify-content-around row"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      id="activateModalOfClasses"
      onClick={[Function]}
      type="button"
    >
      Add New
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      id="uploadFileOfClasses"
      onClick={[Function]}
      type="button"
    >
      Upload File
    </button>
  </div>,
  <ul
    className="list-group list-group-flush"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item"
      data-rb-event-key={null}
      disabled={false}
      id="classesList-header"
      onClick={[Function]}
    >
      <div
        className="d-flex w-100 pt-3 justify-content-between no-gutters"
      >
        <div
          className="col-sm-6"
        >
          <h4>
            Class Name
          </h4>
        </div>
        <div
          className="col-sm-6"
        >
          <h4>
            Max Available Seats
          </h4>
        </div>
      </div>
    </div>
    <div
      className="list-group-item"
      data-rb-event-key={null}
      disabled={false}
      id="class-1"
      onClick={[Function]}
    >
      <div
        className="d-flex w-100 pt-3 justify-content-between no-gutters"
      >
        <div
          className="col-sm-6"
        >
          <h6
            id="nameOfClass_1"
          >
            A1
          </h6>
        </div>
        <div
          className="col-sm-6"
        >
          <h6
            id="seatsOfClass_1"
          >
            70
          </h6>
        </div>
      </div>
    </div>
  </ul>,
]
`;
